### PR TITLE
fix(kernel): recognize client.v1.* connection topics from uplink capsules

### LINKED ISSUE #788: Connection counter is dead: tracker only matches typed Connect/Disconnect, unreachable from capsules
## Problem

The kernel connection tracker (`kernel_router::spawn_connection_tracker`) updates `active_connections` only when it sees a **typed** `IpcPayload::Connect` / `IpcPayload::Disconnect` payload:

```rust
match &message.payload {
    IpcPayload::Disconnect { .. } => kernel.connection_closed(&principal),
    IpcPayload::Connect          => kernel.connection_opened(&principal),
    _ => {}
}
```

But **no capsule can ever produce those variants.** The SDK publish surface is JSON-only — `publish` / `publish_json` / `publish_as` / `publish_json_as` (`sdk-rust/astrid-sdk/src/ipc.rs`) — and never exposes the typed `IpcPayload::Connect`/`Disconnect`. The CLI proxy flattens inbound socket frames to `RawJson` via `publish_json_as`, so even the typed `Disconnect` the CLI client sends over the socket is lost before it reaches the bus.

Net effect: `total_connection_count()` is structurally pinned at **0**. The ephemeral idle-shutdown gate and `astrid who` see no connections regardless of reality. (Several stale comments describe wiring — "the proxy publishes `client.v1.connected`" — that does not exist.)

## Fix (kernel half)

Classify connection lifecycle by **topic** (`client.v1.connected` / `client.v1.disconnect`) in addition to the typed payload, so uplink capsules — which can only publish JSON — can drive the counter. Native producers' typed payloads continue to work.

The capsule-cli companion (separate PR/issue) has the proxy publish those topics carrying the authenticated principal, using stream-close as the authoritative disconnect.

### PR DESCRIPTION
**Pull request by joshuajbouw**
## Linked Issue

Closes #788

## Summary

The kernel connection tracker matched only the typed `IpcPayload::Connect`/`Disconnect` variants — which no capsule can ever produce, because the SDK publish surface is JSON-only and never exposes those variants (and the CLI proxy flattens inbound socket frames to `RawJson`). So uplink capsules could never populate `active_connections`, leaving `total_connection_count()` pinned at 0: the ephemeral idle-shutdown gate and `astrid who` saw no connections regardless of reality. This is the kernel half of the fix.

## Changes

- Add a pure, unit-tested `connection_signal(topic, payload)` helper that classifies a `client.v1.*` message as open/close using **both** the typed payload (native producers) **and** the topic (`client.v1.connected` / `client.v1.disconnect`, for uplink capsules that can only publish JSON).
- Rewire `spawn_connection_tracker` to dispatch via the helper.
- Correct the stale comments that described the non-existent "proxy publishes `client.v1.connected`" wiring.
- CHANGELOG entry under `[Unreleased] / Fixed`.

### Automated

- [x] `cargo test --workspace` passes
- [x] No new clippy warnings (`cargo clippy -p astrid-kernel --all-features -- -D warnings` clean)

7 new unit tests in `connection_tracker_tests.rs`: typed Connect/Disconnect (with and without reason), topic-only JSON from capsules (connected/disconnect), unrelated `client.v1.*` ignored, and typed-payload-wins-on-mismatched-topic.

### Manual (optional)

End-to-end verification depends on the capsule-cli companion (below) that publishes the topics. This PR is backward-compatible and safe to land independently.

## Checklist

- [x] Linked to an issue
- [x] CHANGELOG.md updated under `[Unreleased]`

---

**Companion change (separate PR, capsule-cli):** the CLI proxy publishes `client.v1.connected`/`client.v1.disconnect` carrying the authenticated principal, tracking per-stream principal, with **stream-close as the authoritative disconnect** (robust to ungraceful exit). This kernel PR is the half that makes those topics actually count.

**Persistent-mode note:** per design (confirmed with the maintainer), persistent `astrid start` daemons must always persist (multi-tenant host); this PR does not change that — it fixes the *signal* (`astrid who`, ephemeral idle), not persistent lifecycle, which stays operator-driven via `KernelRequest::Shutdown`.

### CODE REVIEW

File: crates/astrid-kernel/src/kernel_router/mod.rs [hunk 1]
```diff
@@ -84,11 +89,36 @@ pub(crate) fn spawn_kernel_router(kernel: Arc<crate::Kernel>) -> tokio::task::Jo
...
+/// the idle monitor / `astrid who` see zero connections regardless of reality.
+fn connection_signal(topic: &str, payload: &IpcPayload) -> Option<ConnectionSignal> {
+    if matches!(payload, IpcPayload::Disconnect { .. }) || topic == "client.v1.disconnect" {
+        Some(ConnectionSignal::Closed)
+    } else if matches!(payload, IpcPayload::Connect) || topic == "client.v1.connected" {
+        Some(ConnectionSignal::Opened)
+    } else {
+        None
+    }
+}
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-kernel/src/kernel_router/mod.rs:132]: Done in 21f1e9c. `ConnectionSignal::Closed` now carries `reason: Option<String>`, extracted from both the typed `IpcPayload::Disconnect { reason }` and a JSON `"reason"` key on the `client.v1.disconnect` topic. Consolidated into a single `match` with typed-payload precedence over the topic.

File: crates/astrid-kernel/src/kernel_router/mod.rs [hunk 2]
```diff
@@ -107,16 +137,16 @@ fn spawn_connection_tracker(kernel: Arc<crate::Kernel>) -> tokio::task::JoinHand
...
                 },
-                IpcPayload::Connect => {
+                Some(ConnectionSignal::Opened) => {
                     kernel.connection_opened(&principal);
-                    debug!(%principal, "New client connection accepted");
+                    debug!(%principal, topic = %message.topic, "New client connection accepted");
                 },
-                _ => {},
+                None => {},
             }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-kernel/src/kernel_router/mod.rs:166]: Done in 21f1e9c. The tracker matches `Closed { reason }` and logs `?reason` again, restoring the original diagnostic.

File: crates/astrid-kernel/src/kernel_router/connection_tracker_tests.rs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+#[test]
+fn typed_disconnect_without_reason_closes() {
+    let payload = IpcPayload::Disconnect { reason: None };
+    assert_eq!(
+        connection_signal("client.v1.disconnect", &payload),
+        Some(ConnectionSignal::Closed)
+    );
+}
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-kernel/src/kernel_router/connection_tracker_tests.rs:38]: Done in 21f1e9c. Updated to assert `Closed { reason: Some("quit") }` and `Closed { reason: None }`.

File: crates/astrid-kernel/src/kernel_router/connection_tracker_tests.rs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+#[test]
+fn disconnect_topic_with_json_payload_closes() {
+    let payload = IpcPayload::RawJson(serde_json::json!({ "reason": "quit" }));
+    assert_eq!(
+        connection_signal("client.v1.disconnect", &payload),
+        Some(ConnectionSignal::Closed)
+    );
+}
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-kernel/src/kernel_router/connection_tracker_tests.rs:70]: Done in 21f1e9c. The JSON-disconnect test asserts the reason is extracted; also added a companion test for a JSON disconnect with no `reason` key (`Closed { reason: None }`).

File: crates/astrid-kernel/src/kernel_router/connection_tracker_tests.rs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    // mismatched topic never suppresses a real connection event.
+    assert_eq!(
+        connection_signal("some.other.topic", &IpcPayload::Connect),
+        Some(ConnectionSignal::Opened)
+    );
+    assert_eq!(
+        connection_signal("some.other.topic", &IpcPayload::Disconnect { reason: None }),
+        Some(ConnectionSignal::Closed)
+    );
+}
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-kernel/src/kernel_router/connection_tracker_tests.rs:93]: Done in 21f1e9c. Updated to `Closed { reason: None }`.
